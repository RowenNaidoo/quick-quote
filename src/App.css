.App {
  margin: auto;
  padding: 0 15px;
}

@media (min-width: 768px) {
  .App {
    width: 768px;
    margin: auto;
    display: flex;
    flex-direction: column;
  }
}
