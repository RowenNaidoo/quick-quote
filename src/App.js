import React from 'react'
import { Route, Switch } from 'react-router-dom'
import styled from 'styled-components'

import QuoteForm from './containers/QuoteForm'
import QuoteResult from './containers/QuoteResult'
import breakPoints from './utils/breakpoints'

function App() {
  return (
    <Container>
      <Header>Quick Quote</Header>
      <ContentContainer>
        <Switch>
          <Route path="/" component={QuoteForm} exact />
          <Route path="/result/:fromCurrency/:toCurrency/:amount" component={QuoteResult} />
        </Switch>
      </ContentContainer>
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto;
  padding: 0 15px;

  @media (min-width: ${breakPoints.tablet}) {
    width: 768px;
    margin: auto;
    display: flex;
    flex-direction: column;
  }
`
const Header = styled.h1`
  flex: 1 1 auto;
  border-bottom: 1px solid lightGray;
  font-weight: normal;
`
const ContentContainer = styled.div`
  background-color: #F3F6F1;
  padding: 0 15px;
`
export default App
