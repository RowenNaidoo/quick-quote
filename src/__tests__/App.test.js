import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import App from '../App'

describe('<App />', () => {
  let component
  const router = {
    history: new BrowserRouter().history,
    route: {
      location: {},
      match: {},
    },
  }

  const createContext = () => ({
    context: { router },
    childContextTypes: { router: {} },
  })

  beforeEach(() => {
    component = shallow(<App />, createContext())
  })

  test('matches snapshot', () => {
    component = renderer.create(<BrowserRouter><App /></BrowserRouter>)
    expect(component.toJSON()).toMatchSnapshot()
  })
});
