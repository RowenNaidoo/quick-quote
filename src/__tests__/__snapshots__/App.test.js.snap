// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="sc-fzoyTs dgFpdj"
>
  <h1
    className="sc-fzoNJl kBeGYw"
  >
    Quick Quote
  </h1>
  <div
    className="sc-fzoXWK kCGSSN"
  >
    <div
      className="sc-fzqBZW IpDVS"
    >
      <form>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoyAV fTvrOA"
          >
            <div
              className="sc-AxjAm iPShpD"
            >
              <div
                className="sc-AxirZ bqmkbb"
              >
                First Name
                <span
                  className="sc-AxhUy enGJvm"
                >
                  *
                </span>
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <input
                  className="sc-AxiKw jgecmR"
                  id="firstName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <span
                  className="sc-AxhCb jNwDnw"
                >
                  
                </span>
              </div>
            </div>
          </div>
          <div
            className="sc-fzoyAV ctfxxU"
          >
            <div
              className="sc-AxjAm iPShpD"
            >
              <div
                className="sc-AxirZ bqmkbb"
              >
                Last Name
                <span
                  className="sc-AxhUy enGJvm"
                >
                  *
                </span>
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <input
                  className="sc-AxiKw jgecmR"
                  id="lastName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <span
                  className="sc-AxhCb jNwDnw"
                >
                  
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoyAV bswOFA"
          >
            <div
              className="sc-AxjAm iPShpD"
            >
              <div
                className="sc-AxirZ bqmkbb"
              >
                Email
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <input
                  className="sc-AxiKw jgecmR"
                  id="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <span
                  className="sc-AxhCb jNwDnw"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoyAV bswOFA"
          >
            <div
              className="sc-fzoLsD iAaohI"
            >
              <div
                className="sc-fzpans cqeKxx"
              >
                Telephone / Mobile
              </div>
              <div
                className="sc-fzpans cqeKxx"
              >
                <div
                  className="sc-fzplWN gFyNBt"
                >
                  <div
                    className="sc-AxgMl bKFcdD"
                  >
                    <div
                      className="sc-AxheI cmsLbY"
                    />
                    <div
                      className="sc-AxheI cmsLbY"
                    >
                      <select
                        className="sc-Axmtr duYEwY"
                        id="contact_dropDown"
                        onChange={[Function]}
                        value=""
                      >
                        <option
                          value=""
                        >
                          Select...
                        </option>
                        <option
                          value="+61"
                        >
                          +61
                        </option>
                      </select>
                    </div>
                    <div
                      className="sc-AxheI cmsLbY"
                    >
                      <span
                        className="sc-AxmLO yHMjz"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="sc-fznyAO eqTdrq"
                >
                  <div
                    className="sc-AxjAm iPShpD"
                  >
                    <div
                      className="sc-AxirZ bqmkbb"
                    />
                    <div
                      className="sc-AxirZ bqmkbb"
                    >
                      <input
                        className="sc-AxiKw jgecmR"
                        id="contact_text"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      className="sc-AxirZ bqmkbb"
                    >
                      <span
                        className="sc-AxhCb jNwDnw"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-fzpans cqeKxx"
              >
                <span
                  className="sc-fznKkj WVcGc"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoyAV fTvrOA"
          >
            <div
              className="sc-AxgMl bKFcdD"
            >
              <div
                className="sc-AxheI cmsLbY"
              >
                From Currency
                <span
                  className="sc-fzozJi krkzdl"
                >
                  *
                </span>
              </div>
              <div
                className="sc-AxheI cmsLbY"
              >
                <select
                  className="sc-Axmtr kVVFsY"
                  id="fromCurrency"
                  onChange={[Function]}
                >
                  <option
                    value=""
                  >
                    Select...
                  </option>
                  <option
                    value="AUD"
                  >
                    Australian Dollar(AUD)
                  </option>
                  <option
                    value="USD"
                  >
                    US Dollar(USD)
                  </option>
                </select>
              </div>
              <div
                className="sc-AxheI cmsLbY"
              >
                <span
                  className="sc-AxmLO yHMjz"
                >
                  
                </span>
              </div>
            </div>
          </div>
          <div
            className="sc-fzoyAV ctfxxU"
          >
            <div
              className="sc-AxgMl bKFcdD"
            >
              <div
                className="sc-AxheI cmsLbY"
              >
                To Currency
                <span
                  className="sc-fzozJi krkzdl"
                >
                  *
                </span>
              </div>
              <div
                className="sc-AxheI cmsLbY"
              >
                <select
                  className="sc-Axmtr kVVFsY"
                  id="toCurrency"
                  onChange={[Function]}
                >
                  <option
                    value=""
                  >
                    Select...
                  </option>
                  <option
                    value="AUD"
                  >
                    Australian Dollar(AUD)
                  </option>
                  <option
                    value="USD"
                  >
                    US Dollar(USD)
                  </option>
                </select>
              </div>
              <div
                className="sc-AxheI cmsLbY"
              >
                <span
                  className="sc-AxmLO yHMjz"
                >
                  
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoyAV bswOFA"
          >
            <div
              className="sc-AxjAm iPShpD"
            >
              <div
                className="sc-AxirZ bqmkbb"
              >
                Amount
                <span
                  className="sc-AxhUy enGJvm"
                >
                  *
                </span>
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <input
                  className="sc-AxiKw jgecmR"
                  id="amount"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="sc-AxirZ bqmkbb"
              >
                <span
                  className="sc-AxhCb jNwDnw"
                >
                  
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-fzqNJr czDRjm"
        >
          <div
            className="sc-fzoLag hDfRZy"
          >
            <button
              className="sc-fzokOt fMtEza"
              onClick={[Function]}
            >
              GET QUOTE
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
